src/wallet/wallet2.cpp

```diff
@@ -6033,6 +6053,35 @@ void wallet2::get_outs(std::vector<std::vector<tools::wallet2::get_outs_entry>>
     COMMAND_RPC_GET_OUTPUTS_BIN::request req = AUTO_VAL_INIT(req);
     COMMAND_RPC_GET_OUTPUTS_BIN::response daemon_resp = AUTO_VAL_INIT(daemon_resp);
 
+    struct gamma_engine
+    {
+      double min() const { return 0.0; }
+      double max() const { return 1.0; }
+      double operator()() { return crypto::rand<uint64_t>() / (double)std::numeric_limits<uint64_t>::max(); }
+    } engine;
```

Comment:
This gives the following [compile error](https://build.getmonero.org/builders/monero-static-osx-10.13/builds/725/steps/compile/logs/stdio): 
```
In file included from /Users/buildbot/monero-builder/monero-static-osx-10_13/build/src/wallet/wallet2.cpp:32:
/Library/Developer/CommandLineTools/usr/include/c++/v1/random:3643:51: error: call to non-static member function without an object argument
    const size_t __logR = __log2<uint64_t, _URNG::max() - _URNG::min() + uint64_t(1)>::value;
                                           ~~~~~~~^~~
/Library/Developer/CommandLineTools/usr/include/c++/v1/random:4167:30: note: in instantiation of function template specialization 'std::__1::generate_canonical<double, 53, gamma_engine>' requested here
                      _VSTD::generate_canonical<result_type,
                             ^
...
```
because the argument to the distribution sampler needs to satisfy the [UniformRandomBitGenerator](https://en.cppreference.com/w/cpp/named_req/UniformRandomBitGenerator) requirements. The following eliminates the error:
```c++
    struct gamma_engine
    {
      typedef uint64_t result_type;
      static constexpr uint64_t min() { return 0; }
      static constexpr uint64_t max() { return std::numeric_limits<uint64_t>::max(); }
      uint64_t operator()() { return crypto::rand<uint64_t>(); }
    } engine;
```